packages/kit/src/exports/vite/graph_analysis/index.js

```diff
@@ -1,277 +1,81 @@
-
-	/** @type {Set<string>} */
-	#seen;
-
-	/**
-	 * @param {(id: string) => import('rollup').ModuleInfo | null} node_getter
-	 * @param {import('rollup').ModuleInfo} node
-	 */
-	constructor(node_getter, node) {
-		this.#node_getter = node_getter;
-		this.#module_info = node;
-		this.id = remove_query_from_id(normalizePath(node.id));
-		this.dynamic = false;
-		this.#seen = new Set();
-	}
-
-	/**
-	 * @param {(id: string) => import('rollup').ModuleInfo | null} node_getter
-	 * @param {import('rollup').ModuleInfo} node
-	 * @param {boolean} dynamic
-	 * @param {Set<string>} seen;
-	 * @returns {RollupImportGraph}
-	 */
-	static #new_internal(node_getter, node, dynamic, seen) {
-		const instance = new RollupImportGraph(node_getter, node);
-		instance.dynamic = dynamic;
-		instance.#seen = seen;
-		return instance;
-	}
-
-	get children() {
-		return this.#children();
-	}
-
-	*#children() {
-		if (this.#seen.has(this.id)) return;
-		this.#seen.add(this.id);
-		for (const id of this.#module_info.importedIds) {
-			const child = this.#node_getter(id);
-			if (child === null) return;
-			yield RollupImportGraph.#new_internal(this.#node_getter, child, false, this.#seen);
-		}
-		for (const id of this.#module_info.dynamicallyImportedIds) {
-			const child = this.#node_getter(id);
-			if (child === null) return;
-			yield RollupImportGraph.#new_internal(this.#node_getter, child, true, this.#seen);
+			for (const child of module.dynamicallyImportedIds) {
+				follow(child, [...chain, { id, dynamic: true }]);
+			}
 		}
-	}
-}
-
-/** @implements {ImportGraph} */
-export class ViteImportGraph {
-	/** @type {Set<string>} */
-	#module_types;
-
-	/** @type {import('vite').ModuleNode} */
-	#module_info;
 
-	/** @type {string} */
-	id;
-
-	/** @type {Set<string>} */
-	#seen;
-
-	/**
-	 * @param {Set<string>} module_types Module types to analyze, eg '.js', '.ts', etc.
-	 * @param {import('vite').ModuleNode} node
-	 */
-	constructor(module_types, node) {
-		this.#module_types = module_types;
-		this.#module_info = node;
-		this.id = remove_query_from_id(normalizePath(node.id ?? ''));
-		this.#seen = new Set();
+		safe.add(id);
 	}
 
-	/**
-	 * @param {Set<string>} module_types Module types to analyze, eg '.js', '.ts', etc.
-	 * @param {import('vite').ModuleNode} node
-	 * @param {Set<string>} seen
-	 * @returns {ViteImportGraph}
-	 */
-	static #new_internal(module_types, node, seen) {
-		const instance = new ViteImportGraph(module_types, node);
-		instance.#seen = seen;
-		return instance;
-	}
-
-	get dynamic() {
-		return false;
-	}
-
-	get children() {
-		return this.#children();
-	}
-
-	*#children() {
-		if (this.#seen.has(this.id)) return;
-		this.#seen.add(this.id);
-		for (const child of this.#module_info.importedModules) {
-			if (!this.#module_types.has(path.extname(this.id))) {
-				continue;
-			}
-			yield ViteImportGraph.#new_internal(this.#module_types, child, this.#seen);
+	return {
+		/** @param {string} id */
+		check: (id) => {
+			follow(id, []);
```

Comment:
default is already empty array

```suggestion
			follow(id);
```